### PR TITLE
Discrete uniform and hyper geometric moment

### PR DESCRIPTION
**Pull request by farhanreynaldo**
Add moments and tests for the below distributions as part of https://github.com/pymc-devs/pymc/issues/5078:

* pymc.distributions.Discrete.DiscreteUniform
* pymc.distributions.Discrete.HyperGeometric

### CODE REVIEW

**Comment by farhanreynaldo** [comment]
I am not quite sure why the `HyperGeometric` moment tests are failing, I have checked the calculation manually but it yielded a different result. Is it anything to do with the `good` and `bad` variables?

File: pymc/distributions/discrete.py [hunk 1]
```diff
@@ -920,6 +920,12 @@ def dist(cls, N, k, n, *args, **kwargs):
         n = at.as_tensor_variable(intX(n))
         return super().dist([good, bad, n], *args, **kwargs)
 
+    def get_moment(rv, size, N, k, n):
+        mode = intX(at.floor((n + 1) * (k + 1) / (N + 2)))
```
Comment (ricardoV94) [inline pymc/distributions/discrete.py]:
```suggestion
        mode = at.floor((n + 1) * (k + 1) / (N + 2))
```

Should be fine without intX, we take care of that in the method that calls `get_moment`

**Comment by ricardoV94** [comment]
> I am not quite sure why the `HyperGeometric` moment tests are failing, I have checked the calculation manually but it yielded a different result. Is it anything to do with the `good` and `bad` variables?

The get_moment should expect whatever is passed to `super().dist` at the end of the `dist` classmethod. That means `get_moment` is getting `good, bad, and n`. You will have to work with those (or convert back to `N, k, n`) if you need them.

File: pymc/distributions/discrete.py (hunk 1 above)
Comment (farhanreynaldo) [reply pymc/distributions/discrete.py]: Sure, I removed it in the latest commit.

**Comment by farhanreynaldo** [comment]
> > I am not quite sure why the `HyperGeometric` moment tests are failing, I have checked the calculation manually but it yielded a different result. Is it anything to do with the `good` and `bad` variables?
> 
> The get_moment should expect whatever is passed to `super().dist` at the end of the `dist` classmethod. That means `get_moment` is getting `good, bad, and n`. You will have to work with those (or convert back to `N, k, n`) if you need them.

So whenever we create a `get_moment` function, the arguments that we passed should match the arguments inside `super().dist`?

**Comment by ricardoV94** [comment]
> > > I am not quite sure why the `HyperGeometric` moment tests are failing, I have checked the calculation manually but it yielded a different result. Is it anything to do with the `good` and `bad` variables?
> > 
> > 
> > The get_moment should expect whatever is passed to `super().dist` at the end of the `dist` classmethod. That means `get_moment` is getting `good, bad, and n`. You will have to work with those (or convert back to `N, k, n`) if you need them.
> 
> So whenever we create a `get_moment` function, the arguments that we passed should match the arguments inside `super().dist`?

Yes, that what will be used when we call get_moment internally

**Comment by farhanreynaldo** [comment]
Is the failing test related to this PR?

**Comment by ricardoV94** [comment]
Probably not, seems to be related to #4771

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
Seems good to me. You have some conflicts that need to be fixed though

**Comment by ricardoV94** [comment]
Thanks @farhanreynaldo

**Pull request merged**